packages/std/src/math.rs

```diff
@@ -0,0 +1,255 @@
+use schemars::JsonSchema;
+use serde::{de, ser, Deserialize, Deserializer, Serialize};
+use std::convert::{TryFrom, TryInto};
+use std::{fmt, ops};
+
+use crate::errors::{generic_err, underflow, StdError, StdResult};
+
+/// Decimal9 represents a fixed-point decimal value with 9 fractional digits.
+/// That is Decimal9(1_000_000_000) == 1
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Decimal9(pub u64);
+
+pub const DECIMAL_FRACTIONAL: u64 = 1_000_000_000;
```

Comment:
I think this can be private (and static?). A user better uses `Decimal9::one().0`

Reply:
I'm not sure the difference between `const` and `static`. I saw both of them mixed together a bit in code, and thought `const ...: u64` was clearer. But fine to change to `static` if there is a reason.

Reply:
Ah, `const` is what we want almost everywhere:

> More specifically, constants in Rust have no fixed address in memory. This is because they’re effectively inlined to each place that they’re used. References to the same constant are not necessarily guaranteed to refer to the same memory address for this reason.

https://doc.rust-lang.org/1.0.0/book/const-and-static.html